HeavyFlavorAnalysis/SpecificDecay/plugins/BPHWriteSpecificDecay.h

```diff
@@ -0,0 +1,175 @@
+#ifndef TestBaseNtuple_h
+#define TestBaseNtuple_h
+
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHAnalyzerTokenWrapper.h"
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHPlusMinusCandidate.h"
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHRecoCandidate.h"
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHTrackReference.h"
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/PatCandidates/interface/GenericParticle.h"
+#include "DataFormats/PatCandidates/interface/CompositeCandidate.h"
+
+#include "RecoVertex/KinematicFitPrimitives/interface/RefCountedKinematicParticle.h"
+#include "RecoVertex/KinematicFitPrimitives/interface/RefCountedKinematicVertex.h"
+
+#include <string>
+#include <vector>
+#include <map>
+#include <iostream>
+#include <fstream>
+
+class TH1F;
+class BPHRecoCandidate;
+
+class BPHWriteSpecificDecay: public BPHAnalyzerWrapper<edm::EDProducer> {
```

Comment:
edm::one::EDProducer 
... just because the code doesn't appear multi-threading friendly.
edm::stream::EDProducer would be a better choice eventually 
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes
... this all applies if this code has any hopes running centrally.